docs/requirements/ADP-957.md

```diff
@@ -0,0 +1,201 @@
+    non-ada assets.
+
+*   The protocol parameter `collateralPercent` is used to specify the
+    percentage of the total transaction fee its collateral must (at a minimum)
+    cover. The collateral inputs must not themselves be locked by a script.
+
+*   The protocol parameter `maxCollateralInputs` is used to limit,
+    additionally, the total number of collateral inputs, and thus the total
+    number of additional signatures that must be checked during validation.
+
+# Requirements
+
+1.  The wallet application, when requested to create a transaction requiring
+    collateral, should attempt to select one or more **pure-ada** entries from
+    the wallet's UTxO set to serve as collateral inputs. The wallet application
+    **must not** select UTxO entries that include assets other than ada to
+    serve as collateral.
+
+2.  The wallet application, when requested to create a transaction requiring
+    collateral, should attempt to select no fewer than one UTxO entry, and no
+    more than `maxCollateralInputs` UTxO entries to serve as collateral.
+
+    If it is not possible to satisfy these bounds, the wallet application must
+    return an error to the caller.
+
+3.  The wallet application, when requested to create a transaction requiring
+    collateral, should attempt to select an amount of ada that is greater than
+    or equal to a minimum bound that is determined by the `collateralPercent`
+    protocol parameter and the transaction fee:
+
+        collateralAmount ≥ txFee × collateralPercent
+
+    If it is not possible to satisfy the minimum bound, the wallet application
+    must return an error to the caller.
+
+4.  The wallet application, when requested to create a transaction requiring
+    collateral, should select an amount of ada that is less than or equal to a
+    maximum bound that is determined by a user-specified `maxCollateralFactor`
+    parameter:
+
+        collateralAmount ≤ txFee × collateralPercent × maxCollateralFactor
+
+    where:
+
+        maxCollateralFactor ≥ 1
+
+    If it is not possible to satisfy the maximum bound, the wallet application
+    must return an error to the caller.
+
+5.  The wallet application may provide a default value for the
+    `maxCollateralFactor` parameter.
+
+6.  The wallet application, when requested to create a transaction requiring
+    collateral, should make a reasonable attempt to select the smallest amount
+    of ada that satisfies the minimum and maximum bounds, but is permitted to
+    select a higher, non-optimal amount (still within bounds) in cases where
+    searching for the optimal solution would entail an unreasonably high
+    performance cost.
+
+7.  The wallet application must ensure that UTxO entries used for collateral
+    inputs in pending transactions are not available for reuse as collateral
+    inputs in future transactions.
+
+8.  The wallet application should consider a UTxO entry used for collateral in
```

Comment:
I know we were talking about this yesterday, but I didn't quite catch the conclusion. Surely if a UTxO is put up for collateral, it shouldn't be made available to selection? Unless we enforce some other property:

Given,
1. some transaction with a collateral input
2. some transaction whose input is the collateral of Tx#1

Tx#2 remains pending until Tx#1 is approved?

Or does the node layer handle this? Like "first-in wins".

Reply:
Thanks for this document btw, I think it's great :+1: